examples/traefik-new/docker-compose-traefik.yml

```diff
@@ -0,0 +1,107 @@
+      - target: 443
+        published: 443
+      - target: 222
+        published: 222
+    deploy:
+      placement:
+        constraints:
+          # Make the traefik service run only on the node with this label
+          # as the node with it has the volume for the certificates
+          - node.labels.traefik-public.traefik-public-certificates == true
+      labels:
+        # Enable Traefik for this service, to make it available in the public network
+        - traefik.enable=true
+        # Use the traefik-public network (declared below)
+        - traefik.docker.network=traefik-public
+        - traefik.constraint-label=traefik-public
+        # admin-auth middleware with HTTP Basic auth
+        - traefik.http.middlewares.admin-auth.basicauth.usersfile=/auth/.htpasswd
+        # https-redirect middleware to redirect HTTP to HTTPS
+        # It can be re-used by other stacks in other Docker Compose files
+        - traefik.http.middlewares.https-redirect.redirectscheme.scheme=https
+        - traefik.http.middlewares.https-redirect.redirectscheme.permanent=true
+        # traefik-http set up only to use the middleware to redirect to https
+        # Uses the environment variable DOMAIN
+        - traefik.http.routers.all-http.rule=HostRegexp(`{host:.+}`)
+        - traefik.http.routers.all-http.entrypoints=http
+        - traefik.http.routers.all-http.middlewares=https-redirect
+        # traefik-https the actual router using HTTPS
+        # Uses the environment variable DOMAIN
+        - traefik.http.routers.traefik-public-https.rule=Host(`my-traefik.example.com`) # !!!!CHANGEME!!!! (REQUIRED)
+        - traefik.http.routers.traefik-public-https.entrypoints=https
+        - traefik.http.routers.traefik-public-https.tls=true
+        # Use the special Traefik service api@internal with the web UI/Dashboard
+        - traefik.http.routers.traefik-public-https.service=api@internal
+        # Use the "le" (Let's Encrypt) resolver created below
+        - traefik.http.routers.traefik-public-https.tls.certresolver=le
+        # Enable HTTP Basic auth, using the middleware created above
+        - traefik.http.routers.traefik-public-https.middlewares=admin-auth
+        # Define the port inside of the Docker service to use
+        - traefik.http.services.traefik-public.loadbalancer.server.port=8080
+    volumes:
+      # Add Docker as a mounted volume, so that Traefik can read the labels of other services
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - <</PATH/TO/YOUR/HTPASSWD>>:/auth/.htpasswd:ro  # !!!!CHANGEME!!!!  (REQUIRED)
+      # Mount the volume to store the certificates
+      - traefik-public-certificates_new:/certificates
+      - <</PATH/TO/YOUR/TRAEFIK/CONFIG/FOLDER/conf.d>>:/conf.d # # !!!!CHANGEME!!!! (OPTIONAL) use for dynamic conf. Remove if not needed.
+    command:
+      # - --log.level=DEBUG
+      # Enable Docker in Traefik, so that it reads labels from Docker services
+      - --providers.docker
```

Comment:
In traefik v3, the `docker` provider has been deprecated for `swarm`.  Please update all references as appropriate.

Reply:
@jnovack I'll make sure a proper v2 --> v3 migration and will let you know when done.